overlord/ifacestate/udevmonitor/udevmon.go

```diff
@@ -131,21 +157,30 @@ func (m *Monitor) udevEvent(ev *netlink.UEvent) {
 }
 
 func (m *Monitor) addDevice(kobj string, env map[string]string) {
-	di, err := hotplug.NewHotplugDeviceInfo(env)
+	dev, err := hotplug.NewHotplugDeviceInfo(env)
 	if err != nil {
 		return
 	}
+	if _, seen := m.devices[dev.DevicePath()]; seen {
```

Comment:
Since `m.devices` is a `map[string]bool` you can just do:
```
if m.devices[dev.DevicePath()] {
   return
}
```

Reply:
Right, done (in all other places below as well).